src/ResourceManager/Search/Commands.Management.Search/SearchService/GetSearchServiceCommand.cs

```diff
@@ -0,0 +1,86 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Management.Search.SearchService
+{
+    [Cmdlet(VerbsCommon.Get, SearchServiceNounStr), OutputType(typeof(Models.PSSearchService))]
+    public class GetSearchServiceCommand : SearchServiceBaseCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceGroupParameterSetName,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = ResourceGroupHelpMessage)]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceNameParameterSetName,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = ResourceGroupHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter()]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceNameParameterSetName,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = ResourceNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSetName,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = ResourceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter()]
+        public string ResourceId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if(ParameterSetName.Equals(ResourceGroupParameterSetName, StringComparison.InvariantCulture))
+            {
+                var svcs = SearchClient.Services.ListByResourceGroupWithHttpMessagesAsync(ResourceGroupName).Result;
+                WriteSearchServiceList(svcs.Body);
+                return;
+            }
+            else if(ParameterSetName.Equals(ResourceIdParameterSetName, StringComparison.InvariantCulture))
+            {
+                var id = new ResourceIdentifier(ResourceId);
+                ResourceGroupName = id.ResourceGroupName;
+                Name = id.ResourceName;
+            }
+
+            try
+            {
+                var svc = SearchClient.Services.GetWithHttpMessagesAsync(ResourceGroupName, Name).Result;
+                WriteSearchService(svc.Body);
+            }
+            catch(Exception e)
+            {
+                // the method throws an exception when there the service does not exist.
```

Comment:
Can you double check that the error thrown is the not found exception, and throw if it is not?